stan/math/prim/arr/fun/promoter.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef STAN_MATH_PRIM_ARR_FUN_PROMOTER_HPP
+#define STAN_MATH_PRIM_ARR_FUN_PROMOTER_HPP
+
+#include <stan/math/prim/scal/fun/promoter.hpp>
+#include <vector>
+
+namespace stan {
+  namespace math {
+    /**
+     * Struct to hold static function for element-wise type promotion.
+     * This specialization is for vectors of differing types.
+     *
+     * @tparam F type of input element
+     * @tparam T type of output element
```

Comment:
Is input/output same as arg1 and arg2 above?  And is one the result of promoting or something?  Input and output and arg1 and arg2 are hard to follow.